docs(oas): writing up docs on `oas/reducer` and `oas/analyzer` (#900)
readmeio/oas: packages/oas/README.md

Because our `Webhook` class extends `Operation`, every API that's available on the [Operation](#operation) class is available on webhooks.

### Additional Utilities

> Beyond the `Oas`, `Operation`, `Callback` and `Webhook` interfaces the `oas` library also offers additional tooling for analyzing and interacting with OpenAPI definitions.

Reviewer: ```suggestion
Beyond the `Oas`, `Operation`, `Callback` and `Webhook` interfaces, the `oas` library also offers additional tooling for analyzing and troubleshooting OpenAPI definitions.
```

#### Analyzer

> The analyzer, `oas/analyzer`, allows you to run a set of query analysis on your API definition to answer various questions.

Reviewer: ```suggestion
The analyzer, `oas/analyzer`, allows you to run a set of query analyses on your API definition to understand the complexity of your API definition.
```

| `links` | Does your API use [links](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#link-object)? |
| `style` | Do any parameters in your API require [style](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#user-content-parameterstyle) serialization?
| `polymorphism` | Does your API use [polymorphism](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#composition-and-inheritance-polymorphism) (`anyOf`, `oneOf`, `allOf`)? |
| `serverVariables` | Does your API use [server variables](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#server-variable-object)? |
| `webhooks` | Does your API use [webhooks](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#oasWebhooks)?
| `xml` | Does any parameter or schema in your API use the [XML object](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#xml-object) for declaring how a schema should be treated in XML? |
<!-- prettier-ignore-end -->

#### Reducer

> The reducer, `oas/reducer`, can be used to reduce an OpenAPI definition down to only the information necessary for a speciifc set of tags, paths, or operations. OpenAPI reduction can be helpful if you are working to troubleshoot a very large API definition as with the reducer you can narrow that down to only the data that you care about -- all while still having a fully functional and valid OpenAPI definition.

Reviewer: ```suggestion
The reducer, `oas/reducer`, can be used to reduce an OpenAPI definition down to only the information necessary for a specific set of tags, paths, or operations. OpenAPI reduction can be helpful to isolate and troubleshoot issues with a very large API definition -- all while still having a fully functional and valid OpenAPI definition.
```

```ts
import petstore from '@readme/oas-examples/3.0/json/petstore.json';
import reducer from 'oas/reducer';

// This will reduce the `petstore` API definition down to only operations, and

> // their used schemas, that are a part of the `Store` tag.

Reviewer: ```suggestion
// any referenced schemas, that are a part of the `Store` tag.
```

console.log(reducer(petstore, { tags: ['Store'] }));

// Reduces the `petstore` down to only the `POST /pet` operation.
console.log(reducer(petstore, { paths: { '/pet': ['post'] } });

// You can also select all of the methods of a given path by using the `*`
// wildcard. The resulting reduced API definition here will contain `POST /pet`
// and `PUT /put`.
console.log(reducer(petstore, { paths: { '/pet': ['*'] } });
```

> > **Note**

Reviewer: ```suggestion
> [!NOTE]
```
